docs/source/user_guide_kg_builder.rst

```diff
@@ -0,0 +1,412 @@
+.. _user-guide-kg-builder:
+
+User Guide: Knowledge Graph Builder
+###################################
+
+
+This page provides information about how to create a Knowledge Graph from
+unstructured data.
+
+.. warning::
+
+    This feature is still experimental. API changes and bug fixes are expected.
+
+    It is not recommended to use it in production yet.
+
+
+******************
+Pipeline structure
+******************
+
+A Knowledge Graph (KG) construction pipeline requires a few components:
+
+- A document **parser**: extract text from files (PDFs, ...)
+- A document **chunker**: split the text into smaller pieces of text, manageable by the LLM context window (token limit).
+- A chunk **embeder** (optional): compute and store the chunk embeddings
+- A **schema builder**: provide a schema to ground the LLM extracted entities and relations and obtain an easily navigable KG.
+- An **entity and relation extractor**: extract relevant entities and relations from the text.
+- A **Knowledge Graph writer**: write the identified entities and relations to a Neo4j database.
+
+.. image:: images/kg_builder_pipeline.png
+  :alt: KG Builder pipeline
+
+
+This package contains the interface and implementations for each of these components, which are detailed in the following sections.
+
+To see an end-to-end example of a Knowledge Graph construction pipeline,
+refer to `this example <https://github.com/neo4j/neo4j-genai-python/blob/main/examples/pipeline/kg_builder.py>`_.
+
+**********************************
+Knowledge Graph Builder Components
+**********************************
+
+Below is a list of the different components available in this package, and how to use them.
+
+Each of these components can be run individually:
+
+.. code:: python
+
+    import asyncio
+    # replace `PdfLoader` by a real component name
```

Comment:
I'm not sure what is meant by "real component name" here

Reply:
Will remove it, I used a non existing component at the beginning but found it more useful to choose one real component instead.